contrib/babelfishpg_tsql/src/pl_exec-2.c

```diff
@@ -1299,6 +1325,16 @@ exec_stmt_exec(PLtsql_execstate *estate, PLtsql_stmt_exec *stmt)
 								 GUC_ACTION_SAVE, true, 0, false);
 		SetCurrentRoleId(current_user_id, false);
 	}
+	
+	/*
+	 * Restore plan_cache_mode
+	 */
+	if ((stmt->exec_with_recompile || created_with_recompile) && !old_plan_mode)
+	{
+		(void) set_config_option("plan_cache_mode", old_plan_mode,
```

Comment:
I don't think this code path is being executed because this is outside of TRY/CATCH block and we re-throw inside the CATCH block.

Reply:
We are following the same logic here as for setting/resetting the "search_path" GUC.